modules/jpms-utils/src/main/java11/org/mule/runtime/jpms/api/JpmsUtils.java

```diff
@@ -333,6 +334,15 @@ public static ModuleLayer createModuleLayer(URL[] modulePathEntries, ClassLoader
     return controller.layer();
   }
 
+  private static Stream<Module> getParentLayersModules(ModuleLayer moduleLayer) {
+    Stream<Module> modules = moduleLayer.modules().stream();
+    for (ModuleLayer parent : moduleLayer.parents()) {
+      modules = Stream.concat(modules, getParentLayersModules(parent));
```

Comment:
this concat may be confusing... bette usa and return a `Set`?

Reply:
Agreed.